components/dashboard/src/data/prebuilds/prebuild-query.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useQuery } from "@tanstack/react-query";
+import { configurationClient, prebuildClient, stream } from "../../service/public-api";
+import { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { Prebuild } from "@gitpod/public-api/lib/gitpod/v1/prebuild_pb";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+
+export function usePrebuildAndConfigurationQuery(prebuildId: string) {
+    return useQuery<{ prebuild?: Prebuild; configuration?: Configuration }, Error>(
+        prebuildQueryKey(prebuildId),
+        async () => {
+            const prebuild = await prebuildClient.getPrebuild({ prebuildId }).then((d) => d.prebuild);
+            let configuration: Configuration | undefined;
+            if (prebuild?.configurationId) {
+                configuration = await configurationClient
+                    .getConfiguration({ configurationId: prebuild.configurationId })
+                    .then((d) => d.configuration);
+            }
+            return {
+                prebuild,
+                configuration,
+            };
+        },
+        {
+            retry: 1,
```

Comment:
We don't need to retry, maybe change it to `retry: false`

Reply:
I think it's fine to leave here. Feel like it's worth the alternative of a random error state.